module/modmanager/manager.go

```diff
@@ -1113,3 +1123,31 @@ func getModuleDataParentDirectory(options modmanageroptions.Options) string {
 	}
 	return filepath.Join(options.ViamHomeDir, parentModuleDataFolderName, robotID)
 }
+
+// GetFailedModules returns the names of all failing modules.
+func (mgr *Manager) GetFailedModules() []string {
```

Comment:
style: should not have Get in the method name https://go.dev/doc/effective_go#Getters